microsoft/testsuites/mdatp/mdatp.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import pathlib
+
+from lisa import Node, SkippedException, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.operating_system import Linux
+from lisa.sut_orchestrator import AZURE
+from lisa.testsuite import simple_requirement
+
+
+@TestSuiteMetadata(
+    area="mdatp",
+    category="functional",
+    description="""
+        Tests to check the behavior of the virtual cdrom device in VMs.
+    """,
+)
+class MdatpSuite(TestSuite):
+    @TestCaseMetadata(
+        description="""
+           Check for Defender installs and dump config info
+        """,
+        priority=3,
+        requirement=simple_requirement(
+            supported_os=[Linux], supported_platform_type=[AZURE]
+        ),
+    )
+    def verify_mdatp_config(self, node: Node) -> None:
+        # check bash is available
+        if not node.execute("command -v bash", shell=True, sudo=True):
+            raise SkippedException("No bash installation was found! Skipping test...")
+
+        # collect some paths before we start the test
+        local_path = pathlib.PurePath(__file__).parent.joinpath("check-mdatp.sh")
+        working_path = node.get_working_path()
+        node_script_path = working_path.joinpath("check-mdatp.sh")
+
+        # copy the bash script to the node
+        node.shell.copy(local_path=local_path, node_path=node_script_path)
+        result = node.execute(
+            cmd=f"chmod +x {str(node_script_path)}",
+            shell=True,
+            sudo=True,
+        )
+        # and run the script to check and dump defender info
+        result = node.execute(
+            cmd=f"bash +x {str(node_script_path)}",
+            shell=True,
+            sudo=True,
+        )
+        script_output = result.stdout.strip()
+        exit_code = result.exit_code
+        if exit_code != 0:
+            raise SkippedException("Did not find mdatp installation on this VM.")
+
+        node.log.info("Found mdatp installation!")
+        node.log.info(f"Found config information: {script_output}")
```

Comment:
From the logic, it looks the test case always passed or skipped. It doesn't provide useful information.

Reply:
> From the logic, it looks the test case always passed or skipped
This is true
>  It doesn't provide useful information.
I disagree. Right now this test either skips because no config info is present or passes if config info is found. We can use this to look for images which ship with mdatp onboarding or config info. Right now we don't know if this is a bug or not, Lili and Baihua and I are on a chat with that team. When we have enough info to validate the info, we'll add a fail condition where if there is config info and it's unexpected, we fail the test.

Reply:
You can define a test variable. If it's enforced and not found, the test case will fail. For example, "verify_mdatp_config_enable_assert".

Reply:
I don't think we need it, for now this should always result in a fail if info is present.